mlinspect/backends/_iter_creation.py

```diff
@@ -111,20 +112,21 @@ def iter_input_annotation_output_join(inspection_count, x_data, x_annotations, y
 
     x_before_with_annotations = pandas.concat([x_data.reset_index(drop=True), x_annotations], axis=1)
     y_before_with_annotations = pandas.concat([y_data.reset_index(drop=True), y_annotations], axis=1)
-    df_x_output = pandas.merge(x_before_with_annotations, output, left_on="mlinspect_index_x",
-                               right_on="mlinspect_index_x", suffixes=["_x", "_output"])
-    df_x_output_y = pandas.merge(df_x_output, y_before_with_annotations, left_on="mlinspect_index_y",
-                                 right_on="mlinspect_index_y", suffixes=["_x_output", "_y_output"])
+    df_x_output = output.merge(x_before_with_annotations, on="mlinspect_index_x", suffixes=["_output", "_x"])
```

Comment:
Based on all tests passing, I don't think we have a test yet where these changes are visible. Would it be possible to create one that tests the changes here in `iter_input_annotation_output_join`? I think it should be possible to create such a test using `merge(..., sort=True)` based on the existing test `test_frame_merge` in `test/monkeypatching/test_patch_pandas.py`.

Reply:
Done in cc39e5480935636158017c552119f999682b4d47